### PR TITLE
chore(monitor): Add jobs for the monitor repo

### PR DESCRIPTION
**Pull request by jchauncey**
There are 3 subdirectories within the monitor repo
we need jobs for each of these directories to build the images

### CODE REVIEW

File: jobs/monitor_jobs.groovy [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        githubPush()
+      }
+
+      wrappers {
+        timestamps()
+        colorizeOutput 'xterm'
+        credentialsBinding {
+          string("DOCKER_PASSWORD", "0d1f268f-407d-4cd9-a3c2-0f9671df0104")
+          string("QUAY_PASSWORD", "c67dc0a1-c8c4-4568-a73d-53ad8530ceeb")
+          string("GITHUB_ACCESS_TOKEN", "8e11254f-44f3-4ddd-bf98-2cabcb7434cd")
```
Comment (arschles) [inline jobs/monitor_jobs.groovy:104]: these look like uuids - are they passwords for a robot account or?

Comment (arschles) [inline jobs/monitor_jobs.groovy:104]: nvm - after speaking privately I understand that these are pointers to jenkins secrets

File: jobs/monitor_jobs.groovy [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  [
+    // for each repo, create a <dir.name>-master and <dir.name>-pr job
+    [type: 'master', branch: 'master'],
+    [type: 'pr', branch: '${sha1}'],
+  ].each { Map config ->
+    isMaster = config.type == 'master'
+    isPR = config.type == 'pr'
+
+    name = isMaster ? dir.name : "${dir.name}-pr"
+    downstreamJobName = defaults.testJob[config.type]
```
Comment (vdice) [inline jobs/monitor_jobs.groovy]: Do you want to kick off downstream test job?  (would be `workflow-test(-pr)` running workflow e2e suite)

If there are no tests covering these yet, I'd see we remove the downstream job for now to shore up resources... we can always (and should) add back when tests are added.

File: jobs/monitor_jobs.groovy [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+          # if triggered by pull request plugin, use ghprbActualCommit
+          export ACTUAL_COMMIT="\${ghprbActualCommit}"
+          # if manually triggered, use sha1
+          if [ -z "\${ghprbActualCommit}" ]; then
+            export ACTUAL_COMMIT="\${sha1}"
+          fi
+          echo ACTUAL_COMMIT="\${ACTUAL_COMMIT}" > \${WORKSPACE}/env.properties
+        """.stripIndent().trim()
+
+        // do not run e2e tests for workflow-manager at this time
```
Comment (vdice) [inline jobs/monitor_jobs.groovy]: see comment above about whether or not it is desired to kick off workflow e2e job.  (If it still is desired, we can remove the specific `workflow-manager` check which isn't applicable in this file)

File: jobs/monitor_jobs.groovy (hunk 2 above)
Comment (jchauncey) [inline jobs/monitor_jobs.groovy]: im not sure how we would even test this other than the components actually start.

File: jobs/monitor_jobs.groovy [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+          make bootstrap || true
+
+          export IMAGE_PREFIX=deisci
+          docker login -e="\$DOCKER_EMAIL" -u="\$DOCKER_USERNAME" -p="\$DOCKER_PASSWORD"
+          DEIS_REGISTRY='' make docker-build ${dockerPush}
+          docker login -e="\$QUAY_EMAIL" -u="\$QUAY_USERNAME" -p="\$QUAY_PASSWORD" quay.io
+          DEIS_REGISTRY=quay.io/ make docker-build ${dockerPush}
+
+          # if triggered by pull request plugin, use ghprbActualCommit
```
Comment (vdice) [inline jobs/monitor_jobs.groovy:123]: though not completely obvious here, this logic for setting up the `ACTUAL_COMMIT` is only applicable for any downstream jobs to update commit status -- can remove if we don't have any downstream jobs.

**Comment by vdice** [comment]
Code LGTM - just commented on whether or not the downstream test job is desired, which I'll leave up to you.  (If not needed, can remove code in places I've commented.)

**Comment by jchauncey** [comment]
@vdice updated commit to remove downstream jobs for now

**Pull request merged**